opm/output/eclipse/RestartValue.cpp

```diff
@@ -60,6 +61,43 @@ namespace Opm {
         , aquifer  { std::move(aquifer_arg) }
     {
     }
+    RestartValue::RestartValue(data::Solution sol,
+                               data::Wells wells_arg,
+                               data::GroupAndNetworkValues grp_nwrk_arg,
+                               data::Aquifers aquifer_arg,
+                               int lgr_grid)
+        : RestartValue(std::move(sol), std::move(wells_arg),
+              std::move(grp_nwrk_arg), std::move(aquifer_arg))
+    {
+        // Filter the wells using the lgr_grid parameter.
+        filter_wells_for_lgr(this->wells, lgr_grid);
+    }
+
+    void RestartValue::filter_wells_for_lgr(data::Wells& all_wells, int lgr_grid){
+        std::vector<std::reference_wrapper<const std::string>> wells_to_be_removed;
+
+        auto filterConnections = [](const std::vector<Opm::data::Connection>& connections, int lgr_grid_num) -> bool {
+            for (const auto& conn : connections) {
+                if (conn.lgr_grid == lgr_grid_num) {
+                    return true;
+                }
+            }
+            return false;
+        };
+
+        for (const auto& pair : all_wells) {
+            const std::string& well_name = pair.first;
+            auto& well_data = pair.second;
```

Comment:
You can use ["structured bindings"](https://en.cppreference.com/w/cpp/language/structured_binding.html) to simplify these assignments:
```C++
for (const auto& [well_name, well_data] : all_wells) {
    // ...
}
```